agent/src/main.rs

```diff
@@ -28,6 +46,46 @@ async fn read_hostname() -> Result<String> {
         .to_owned())
 }
 
+async fn read_ssh_pub_host_key(String key_location) -> Result<String> {
+    let public_key = tokio::fs::read_to_string(key_location)
+        .await
+        .context("error reading pub key")?
+        .trim_end()
+        .split(" ")
+        .nth(1)
+        .to_owned()
+    let pub_key_decoded = base64::decode(public_key).unwrap()
+    let mut sha256_hasher = Shae256::new()
+    sha256_hasher.update(pub_key_decoded)
+    let hashed_result = sha256_hasher.finalize()
+    Ok(str::from_utf8(simple_hex(&hashed_result)).chars().filter(|c| !c.is_whitespace()).collect())
+}
+
+fn match_ssh_pub_entry(keys: &HashMap<String, String>, key_type: str, pub_entry: Result<String>) {
+    match pub_entry {
+        Ok(pub_key) =>  keys.insert(key_type.to_string(), pub_key),
+        Err(e) => println!("Error when reading {}_pub_key: {:?}", key_type, e) 
+    }
+} 
+
+async fn read_ssh_host_keys(rsa: String, dsa: String, ecdsa: String, ed25519: String) -> Result<HashMap<String, String>> {
+    let mut keys = HashMap::new();
+
+    let (rsa_entry, dsa_entry, ecdsa_entry, ed25519_entry) = tokio::join!(
+        read_ssh_pub_host_key(rsa),
```

Comment:
This isn't a completely accurate statement of what this function does. It gathers the fingerprint of the key. Needs a better name.